### ISSUE TITLE
[🚀 Feature]: Adding BiDi in DefinitelyTyped

### ISSUE THREAD

**Issue by dtopuzov** [SeleniumHQ/selenium#11670]
### Feature and motivation

First I want to thank you for introducing [Bidi Support](https://github.com/SeleniumHQ/selenium/pull/11395) and [Browsing context commands](https://github.com/SeleniumHQ/selenium/pull/11473).
It will be great if the features above are present in [DefinitelyTyped](https://github.com/TamsilAmani/DefinitelyTyped/tree/master/types/selenium-webdriver) so we can use it in TypeScript projects.

### Usage example

```
    beforeEach(async function () {
      driver = await env
        .builder()
        .setFirefoxOptions(new firefox.Options().enableBidi())
        .build()
    })
```
Do not compile in TS project because `enableBidi()` is not present in DefinitelyTyped.

**Comment by titusfortner** [comment]
@pujagani is this the responsibility of Selenium project, or is this something that anyone can contribute to the project by taking what has already been implemented in our repo and adding it to that other repo?

I'm not sure we should be tracking this here since there will always be things we can put over there. What do you think?

**Comment by pujagani** [comment]
Anyone can help contribute to the project and it might be a good candidate for first-bug sort of work too. You are right, tracking this is hard because tracking Selenium's BiDi work itself is hard and then tracking something that is part of a different project is harder.

**Comment by diemol** [comment]
If I understand this correctly, what goes into DefinitelyTyped is the type definitions for TypeScript. Then, for new code, this should be added by the person adding that code. 

For existing code, we should create issues for missing types. Overall, I think this is the responsibility of the project, or at least point out what is missing. 

Are the docs that show how to add these things?

**Comment by dtopuzov** [comment]
@diemol you are right, DefinitelyTyped is a community effort to provide typings for packages that do not provide such together with the package.

I think it will be better if the Selenium project provides its official typings together with selenium-webdriver npm package.
Hope that will give more consistency between what is available as APIs in JavaScript and what is exposed in typings for TS projects (why not even re-write selenium-webdriver package itself in TypeScript).

Current definitions at DefinitelyTyped are ok in general (no major issues, just some missing pieces like this one).
I hope once we have official BiDi support in Selenium we will have some way to use it in TypeScript projects too.

I think TypeScript support is important for the success of the Selenium project because most modern web development is Typescript (Angular/React/Vue) and engineers prefer writing tests in the same language they implemented the product.

**Comment by titusfortner** [comment]
It's an open issue — https://github.com/SeleniumHQ/selenium/issues/10942
@harsha509 was looking into it at one point, but I know he's got a lot on his plate right now.

@AutomatedTester what does @nightwatchjs do? Is there an interest from your end on changing what we are doing?

**Comment by diemol** [comment]
Why don't we put the type definitions in our repo instead? @harsha509 do we still want to move to TypeScript?

**Comment by harsha509** [comment]
> Why don't we put the type definitions in our repo instead? [@harsha509](https://github.com/harsha509) do we still want to move to TypeScript?

HI @diemol ,

I have already moved entire bindings to TS (except atoms). Currently fixing type issues and migrating tests.

Thanks,
Sri

**Comment by diemol** [comment]
That is great, @harsha509!

**Comment by titusfortner** [comment]
@harsha509 what more needs to be done for this issue, or can we close it?
